src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ArrayModelBinder.cs

```diff
@@ -16,16 +16,17 @@ namespace Microsoft.AspNetCore.Mvc.ModelBinding
     public class ArrayModelBinder<TElement> : CollectionModelBinder<TElement>
     {
         /// <inheritdoc />
-        public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        public override Task BindModelAsync(IModelBindingContext bindingContext)
         {
             if (bindingContext == null)
             {
                 throw new ArgumentNullException(nameof(bindingContext));
             }
+            Debug.Assert(bindingContext.Result == null);
 
             if (bindingContext.ModelMetadata.IsReadOnly)
             {
-                return ModelBindingResult.NoResultAsync;
+                return Internal.TaskCache.CompletedTask;
```

Comment:
No. We do namespace qualification only when there's conflicts. Even you'll see a named `using` almost as often.